### PR TITLE
performance regression suite

### PR DESCRIPTION
**Pull request by maheshambule**
Contributors: @dhruvaray @prashantsail, @maheshambule

## Description of changes:
It adds a performance monitoring test suite. This suite enables to quickly add test cases and specify pass fail criteria for different metrics. It creates a standard XML and HTML report with JUnit standard specifications. Refer README for more details.

Components added:

- Test suite runner script
- Sample test cases
- Server monitoring script
- Local monitoring script
- README

TODO- 
1. Add more metrics including GPU metrics
2. Auto Threshold calculation, per environment threshold
3. Environment profiles
4. Add more tests
5. Enable test case variables such as host, port to be specified from suite runner script

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dhanainme** [comment]
Can you please add the CircleCI logs / screenshots if possible.

**Comment by maheshambule** [comment]
@dhanainme, There is a separate PR circle CI integration. However, we had tested Circle CI and performance suite integration.

Below are the links of Circle C job:

https://app.circleci.com/pipelines/github/prashantsail/multi-model-server/224/workflows/b4180b58-d141-4bef-9776-114763384e4c/jobs/563/steps

HTML Report:
https://563-265769776-gh.circle-artifacts.com/0/tmp/mms-performance-regression/junit.html

**Circle CI Job:**
Circle CI Job

**HTML report:**
html_report

**Comment by dhanainme** [comment]
Follow up on in-person feedback from the call : 

Documentation 

	* Add details on Motivation & Scope of the tests in README.md
	* Cookbook for running test  - Explain config / result / metrics / HTML report for a Single test
 
Simple Code Refactors

	* Verbose explanation of comparison test result - Enabling easy understanding / ability to tune thresholds 
	* Docs for all python classes	
		- Document & split classes to separate files  - specifically - Monitoring Agent / Comparison / Suite
		- Interaction with Taurus APIs  / JMX
		- Refactor / Break down ~150 line compare_artifacts python method

Eliminate Redundancy

	* Eliminate redundancy across tests - consider generating the taurus / jmeter configs on the fly

**Comment by dhanainme** [comment]
This looks good to me. I ll have this merged tomorrow.

File: tests/performance/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+gevent==20.5.2
+junitparser==1.4.1
+git+https://github.com/maheshambule/vjunit.git#egg=vjunit
```
Comment (maaquib) [inline tests/performance/requirements.txt:3]: Just curious, what are we using this for?

Comment (maheshambule) [reply tests/performance/requirements.txt:3]: We are using vjunit for generating HTML report from Junit XML. There was a bug in it so I had fixed in my fork and using it here.

**Pull request merged**